test/sim/moves/shelltrap.js

```diff
@@ -32,4 +32,13 @@ describe('Shell Trap', function () {
 		battle.makeChoices('move shelltrap, move splash', 'move tackle, move splash');
 		assert.equal(move.pp, move.maxpp - 2);
 	});
+
+	it('should not power up if hit by a Z- or Max move', function () {
+		battle = common.createBattle({}, [
+			[{species: 'Turtonator', moves: ['shelltrap']}],
+			[{species: 'Magikarp', moves: ['flail']}],
+		]);
+		battle.makeChoices('move shelltrap', 'move flail dynamax');
+		assert(battle.log.some(line => line.includes('|Shell Trap|')));
```

Comment:
?

`|cant|...|Shell Trap|` and `|move|...|Shell Trap|` both contain `|Shell Trap|`.

Is this because `|move|...|Shell Trap` doesn't? I don't really want to rely on that.

Reply:
It looks like Shell Trap either succeeds or does nothing. Why not just test to see if Magikarp's HP equals its max HP?

Reply:
There's no `cant` in this test?

Reply:
Bah, Git only showed me your first comment.

Reply:
I'm checking whether Shell Trap succeeds or whether it turns in to a successful Max Flare. (But because Max Flare thinks you're using a hacked move, it doesn't set sun, so I've got to test it this way instead.)

Reply:
Oh! I think "power up" is too ambiguous there.